host.py

```diff
@@ -240,17 +240,21 @@ def ssh_connect(self, username: str, password: Optional[str] = None, rsa_path: s
 
     def ssh_connect_looped(self, logins: list[Login]) -> None:
         if len(logins) == 0:
-            raise Exception("No usuable logins found")
+            raise Exception("No usable logins found")
+
+        login_details = ", ".join([repr(login) for login in logins])
+        logger.info(f"Attempting SSH connections with logins: {login_details}")
+
         while True:
```

Comment:
This is unrelated to your PR, do we want to rethink this `while True`loop? Do we want to consider a timeout or a certain number of tries before exiting?

Reply:
It is a good point. I think it currently fails when we hit an authentication exception. I'll change it to 360 tries, which is 1 hour with the 10 second timeout.